### PR TITLE
Add ability to select columns for the last sample replicates

### PR DESCRIPTION
**Pull request by kweav**
⚠️ This is a stacked PR based on PR #38 ⚠️ and should complete the described necessities/additions for 3 parameters as described in PR #35

This parameter for selecting columns doesn't directly impact a filter, but does impact two graphs. So there are only changes to the `R/plots-qc.R` file and the `inst/rmd/gimapQCTemplate.Rmd` file. Specifically impacting the `qc_variance_hist()` and 
 `qc_constructs_countzero_bar()` functions. The changes to these functions include adding to the headers (param and description information), replacing anywhere numbers are hardcoded with the parameter, and sets the parameters to a default if they're null. Within the QC Template Rmd, changes include referencing the parameters in the function calls.

I can confirm that I knit the `getting-started.Rmd` successfully locally. The only oddity is that the heatmap is still showing up in the vignette rather than the QC report (as mentioned in PR #33)  

Also need to setup something to check the identity of the target columns if they are not null as described in PR #38

### CODE REVIEW

File: R/plots-qc.R [hunk 1]
```diff
@@ -103,8 +106,12 @@ qc_variance_hist <- function(gimap_dataset, wide_ar = 0.75){
 }
 
 #' Create a bar graph that shows the number of replicates with a zero count for pgRNA constructs flagged by the zero count filter
-#' @description A short description...
+#' @description This bar graph first uses the specified `filter_zerocount_target_col` columns to flag pgRNA constructs that have a raw count of 0 in any one of those columns/samples of interest.
+#' Then, it looks at the specified columns for the final day/sample replicates (`filter_replicates_target_col`) to see for pgRNAs that were flagged by the filter, how many of those replicate samples had raw counts of zeros. And it produces a bar plot reporting on this.
+#' Note, if you select samples/columns to check with the filter that don't have the replicate samples, this graph won't be informative. So you want there to be overlap between the columns for the two target_col parameters to have an informative graph
```
Comment (cansavvy) [inline R/plots-qc.R:111]: Great notes. This is excellent documentation!

File: R/plots-qc.R [hunk 2]
```diff
@@ -113,18 +120,32 @@ qc_variance_hist <- function(gimap_dataset, wide_ar = 0.75){
 #' @examples \dontrun{
 #' gimap_dataset <- get_example_data("gimap")
 #' qc_constructs_countzero_bar(gimap_dataset)
+#' 
+#' #or if you want to select a specific column(s) for looking at where/which samples zero counts are present for
+#' qc_constructs_countzero_bar(gimap_dataset, filter_zerocount_target_col = c(3:5))
```
Comment (cansavvy) [inline R/plots-qc.R:125]: You shouldn't need to have the `c()` here?

File: R/plots-qc.R [hunk 3]
```diff
@@ -113,18 +120,32 @@ qc_variance_hist <- function(gimap_dataset, wide_ar = 0.75){
...
+  if(is.null(filter_zerocount_target_col)){filter_zerocount_target_col <- c(1:ncol(gimap_dataset$raw_counts))}
+  
+  qc_filter_output <- qc_filter_zerocounts(gimap_dataset, filter_zerocount_target_col = filter_zerocount_target_col)
+  
+  if(is.null(filter_replicates_target_col)){ filter_replicates_target_col <- c((ncol(gimap_dataset$transformed_data$log2_cpm)-2) : ncol(gimap_dataset$transformed_data$log2_cpm))} #last 3 columns of the data
+  
 
   return(
-    example_counts[qc_filter_output$filter, c(3:5)] %>%
+    example_counts[qc_filter_output$filter, filter_replicates_target_col] %>%
```
Comment (cansavvy) [inline R/plots-qc.R:145]:
```suggestion
    gimap_dataset$raw_counts[qc_filter_output$filter, filter_replicates_target_col] %>%
```
I think this is what you mean?

**Review by cansavvy — COMMENTED** [review]
A couple questions here but I think we're moving along!! :tada:

File: R/plots-qc.R (hunk 2 above)
Comment (kweav) [reply R/plots-qc.R:125]: Yeah I think you're right. I just hadn't gone through and removed them from all the PRs following the code review last week.

File: R/plots-qc.R (hunk 3 above)
Comment (kweav) [reply R/plots-qc.R:145]: Thanks for catching that!

File: R/plots-qc.R (hunk 2 above)
Comment (kweav) [reply R/plots-qc.R]: pushed a change that simplifies the ones I found

**Review by cansavvy — APPROVED** [review]
LGTM!

**Pull request merged**